### PR TITLE
update UI

### PR DESCRIPTION
**Pull request by bengsquared**

### CODE REVIEW

**Comment by bengsquared** [comment]
@andrew-asana  can I get a LGTM :)

File: common/TaskAnalytics/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+        return;
+      }
+    };
+    fetchData();
+  }, []);
+
+  return (
+    <div className="px-4 py-2 m-auto my-20 max-w-2xl flex flex-col content-center">
+      <h1>Task Analyics Summary</h1>
```
Comment (andrew-asana) [inline common/TaskAnalytics/index.js:27]: Typo here

File: common/TaskAnalytics/api.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        return reject(result);
+      }
+
+      return resolve(result);
+    });
+  });
+}
+
+const api = async (req, res) => {
+  console.log("here!");
```
Comment (andrew-asana) [inline common/TaskAnalytics/api.js:41]: Do we want to keep this log in?

File: common/TaskAnalytics/api.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  let session;
+  try {
+    session = await getServerSession(req, res, authOptions);
+    console.log("session : ", session);
+  } catch (error) {
+    console.log("error : ", error);
+  }
+
+  await applyCors(req, res, cors);
+  const route = req.query.all;
```
Comment (andrew-asana) [inline common/TaskAnalytics/api.js:51]: is `query` and `all` guaranteed to exist? It maybe, but just want to highlight this to keep it consistent with the other  option chaining on the next few lines.

File: common/TaskAnalytics/api.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                  },
+                });
+              }
+            }
+          )
+          .promise();
+      } catch (error) {
+        console.log("error!");
+        console.log(error);
+        res.status(500).send();
```
Comment (andrew-asana) [inline common/TaskAnalytics/api.js:153]: Do we want to send an empty response or is there something we want to send with the error?

File: common/TaskAnalytics/api.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    <script>
+      window.opener.postMessage("success", "https://app.asana.com");
+      window.close();
+    </script>
+  </body>
+</html>`);
+    } else if (route[1] == "attach") {
+      console.log(req.body);
+      console.log("attach");
+      const dataJson = JSON.parse(req.body.data);
```
Comment (andrew-asana) [inline common/TaskAnalytics/api.js:174]: Should try-catch this one too.

File: common/TaskAnalytics/api.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+                TrackerIDTaskID: TrackerIDTaskID,
+                Counter: 0,
+                TrackerID: TrackerID,
+                TaskID: TaskID,
+              },
+            },
+            function (err, data) {
+              if (err) {
+                console.log(err);
+                res.status(500).send();
```
Comment (andrew-asana) [inline common/TaskAnalytics/api.js:195]: Same question as above - more curious.

File: common/TaskAnalytics/api.js [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+          .promise();
+      } catch (error) {
+        console.log("error!");
+        console.log(error);
+        res.status(500).send();
+        return;
+      }
+    }
+  } else {
+    res.status(200).send();
```
Comment (andrew-asana) [inline common/TaskAnalytics/api.js:216]: Same as above question.

**Review by andrew-asana — CHANGES_REQUESTED** [review]
Super solid - just some minor questions and checks.

File: common/TaskAnalytics/api.js (hunk 3 above)
Comment (bengsquared) [reply common/TaskAnalytics/api.js:51]: they're default members of the request object

Comment (bengsquared) [reply common/TaskAnalytics/api.js:51]: I think we're OK

File: common/TaskAnalytics/api.js (hunk 4 above)
Comment (bengsquared) [reply common/TaskAnalytics/api.js:153]: nah the 500 is an internal server error so not much the user can do here -  would be good to log something nice for all of the errors tho

File: common/TaskAnalytics/api.js (hunk 7 above)
Comment (bengsquared) [reply common/TaskAnalytics/api.js:216]: 200 is success, don't need much

Comment (bengsquared) [reply common/TaskAnalytics/api.js:216]: success no-op

**Pull request merged**